lib/cli/src/Cardano/CLI.hs

```diff
@@ -378,44 +386,54 @@ hoistKeyScheme
 hoistKeyScheme eta s = CliKeyScheme
     { allowedWordLengths = allowedWordLengths s
     , mnemonicToRootKey = eta . mnemonicToRootKey s
+    , deriveChildKey = \k i -> eta $ deriveChildKey s k i
     }
 
-xPrvToHexText :: XPrv -> Either String Text
-xPrvToHexText = left showErr . fmap (T.pack . B8.unpack . hex) . unXPrvStripPub
+-- | Pair of functions representing the bidirectional transformation between
+-- a @XPrv@ and its 96-byte hex-encoded form.
+xPrvToTextTransform :: (XPrv -> Either String Text, Text -> Either String XPrv)
```

Comment:
can't we just use `FromText` and `ToText` of `XPrv` (or probably some wrapped `XPrv` to avoid orphan instances) instances instead?

Reply:
> can't we just use FromText and ToText of XPrv (or probably some wrapped XPrv like we do in

We _could_ probably have gone that route.

I thought this way played nicely with
```Haskell
data CliKeyScheme key m = CliKeyScheme
    { allowedWordLengths :: [Int]
    , mnemonicToRootKey :: [Text] -> m key
    , deriveChildKey :: key -> DerivationIndex -> m key
    }
```
since it
- Encapsulated more logic in `CliKeyScheme` that could be tested easily. It enables writing the `scheme == mapKey (fromHex . toHex) scheme` property.*
- More easily allows multiple encodings (e.g. if wanted to add additional support for bech32 at some point)

*) Although I'm not clear on exactly what value this brings compared two normal `FromText`/`ToText` properties. Might (just) be to ensure the roundtrips are performed with the all XPrvs that are used for the other tests, and not a subset.

Had we supported `Byron`, it would have made sure DerivationScheme1 child keys also round-tripped, without us having to add DerivationScheme1 child-keys to the `Arbitrary XPrv` generator (not an obvious thing to do).